src/apps/filters/h3.c

```diff
@@ -348,6 +348,320 @@ SUBCOMMAND(getIcosahedronFaces,
+        free(distances);
+    }
+    // Now that we have the correct data, however we got it, we can print it out
+    printf("[ \"%" PRIx64 "\"", out[0]);
+    for (int64_t i = 1; i < len; i++) {
+        if (out[i] != 0) {
+            printf(", \"%" PRIx64 "\"", out[i]);
+        }
+    }
+    free(out);
+    printf(" ]");
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(gridPathCells,
+           "Returns a JSON array of H3 cells from the origin cell to the "
+           "destination cell (inclusive)") {
+    H3Index origin = 0;
+    Arg originArg = {.names = {"-o", "--origin"},
+                     .required = true,
+                     .scanFormat = "%" PRIx64,
+                     .valueName = "cell",
+                     .value = &origin,
+                     .helpText = "The origin H3 cell"};
+    H3Index destination = 0;
+    Arg destinationArg = {.names = {"-d", "--destination"},
+                          .required = true,
+                          .scanFormat = "%" PRIx64,
+                          .valueName = "cell",
+                          .value = &destination,
+                          .helpText = "The destination H3 cell"};
+    Arg *args[] = {&gridPathCellsArg, &helpArg, &originArg, &destinationArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    int64_t len = 0;
+    H3Error err = H3_EXPORT(gridPathCellsSize)(origin, destination, &len);
+    if (err) {
+        return err;
+    }
+    H3Index *out = calloc(len, sizeof(H3Index));
+    err = H3_EXPORT(gridPathCells)(origin, destination, out);
+    if (err) {
+        free(out);
+        return err;
+    }
+    printf("[ \"%" PRIx64 "\"", out[0]);
+    for (int64_t i = 1; i < len; i++) {
+        if (out[i] != 0) {
+            printf(", \"%" PRIx64 "\"", out[i]);
+        }
+    }
+    free(out);
+    printf(" ]");
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(gridDistance,
+           "Returns the number of steps along the grid to move from the origin "
+           "cell to the destination cell") {
+    H3Index origin = 0;
+    Arg originArg = {.names = {"-o", "--origin"},
+                     .required = true,
+                     .scanFormat = "%" PRIx64,
+                     .valueName = "cell",
+                     .value = &origin,
+                     .helpText = "The origin H3 cell"};
+    H3Index destination = 0;
+    Arg destinationArg = {.names = {"-d", "--destination"},
+                          .required = true,
+                          .scanFormat = "%" PRIx64,
+                          .valueName = "cell",
+                          .value = &destination,
+                          .helpText = "The destination H3 cell"};
+    Arg *args[] = {&gridDistanceArg, &helpArg, &originArg, &destinationArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    int64_t distance = 0;
+    H3Error err = H3_EXPORT(gridDistance)(origin, destination, &distance);
+    if (err) {
+        return err;
+    }
+    printf("%" PRIx64, distance);
+    return E_SUCCESS;
```

Comment:
do we want to print an error message here in addition to returning the appropriate code?

Reply:
What if I modify the `DISPATCH_SUBCOMMAND` macro to automatically print a message to `stderr` based on the error code provided?

I recall a discussion with @nrabinowitz to have a collection of error code names and descriptions in the C library that bindings could use. If we did that, it would be pretty clean to re-use it here.

Reply:
I've put up an implementation for this, but would really like some feedback on it. I'm not super enthused with the implementation I created but I couldn't figure out a better way (putting the array in the `.h` file caused weird compilation issues, so I unfortunately had to split the definition in this way).